iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/VariantManagerImplTest.java

```diff
@@ -435,4 +435,36 @@ void testVariantIndexSwitch() throws Exception {
         variantManager.allowVariantMultiThreadAccess(false);
         assertEquals(VariantManagerConstants.INITIAL_VARIANT_ID, variantManager.getWorkingVariantId());
     }
+
+    @Test
+    void testRetainedPropertyStateful() {
+        Network network = Network.create("test", "test");
+        Substation s = network.newSubstation()
+                .setId("S")
+                .setCountry(Country.FR)
+                .add();
+        VoltageLevel vl = s.newVoltageLevel()
+                .setId("VL")
+                .setNominalV(400.0)
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .add();
+        Switch b1 = vl.getNodeBreakerView().newBreaker()
+                .setId("B1")
+                .setNode1(0)
+                .setNode2(1)
+                .setOpen(false)
+                .setRetained(true)
+                .add();
+
+        VariantManager variantManager = network.getVariantManager();
+        variantManager.cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, "backup");
```

Comment:
I would keep the multithread access since the test is related to the implementation

```suggestion
        VariantManager variantManager = network.getVariantManager();
        variantManager.allowVariantMultiThreadAccess(true);
        variantManager.cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, "backup");
```

Reply:
I'm not sure we ever wanted to test multiThreadAccess here. I think we only wanted to test that the retained property can be set to a different value on different variant. I added it back too as asked.